### ISSUE TITLE
systemd 243.0-1 breaks light-locker on second lock activation

### ISSUE THREAD

**Issue by chrishoage** [the-cavalry/light-locker#146]
This is an odd one. Recently light-locker has started to not send me to the greeter only after my second time locking. The first time works as normal. The second time I am presented with the lock screen and never sent to VT8. If I manually go to VT8 nothing is there. 

To fix it I have to go to VT2 and `pkill light-locker` to get my session back. 

It doesn't appear the greeter matters here, I've tried with lightdm-mini-greeter and webkit2-greeter and both have this issue. 

Lines of note are ones relationg to session switching 
```
[gs_manager_timed_switch] gs-manager.c:445 (00:09:20):	 Start switch to greeter timer
```
None of these kinds of logs that are present in the first lock attempt are present in the second lock attempt. 

A systemd update was installed directly before these issues happened. I noticed some systemd references in the logs in the first lock attempt that were not present int he second lock attempt
```
[listener_dbus_handle_system_message] gs-listener-dbus.c:1374 (00:09:22):	 systemd notified ActiveSession 0
```
systemd update in question, in case others have similar issues. When I have time later in this week I will attempt a downgrade and see if it has any effect with my issue. 
```
[2019-09-08 19:45] [ALPM] upgraded systemd (242.84-2 -> 243.0-1)
[2019-09-08 19:45] [ALPM] upgraded systemd-sysvcompat (242.84-2 -> 243.0-1)
```

#### full debug logs
```
[gs_debug_init] gs-debug.c:106 (00:09:02):	 Debugging enabled
[main] light-locker.c:142 (00:09:02):	 initializing light-locker 1.8.0
[main] light-locker.c:164 (00:09:02):	 Platform:
gtk:        3
systemd:    yes
ConsoleKit: no
UPower:     no
[main] light-locker.c:196 (00:09:02):	 Features:
lock-after-screensaver: yes
late-locking:           yes
lock-on-suspend:        yes
lock-on-lid:            no
settings backend:       GSETTINGS
[main] light-locker.c:198 (00:09:02):	 lock after screensaver 5
[main] light-locker.c:199 (00:09:02):	 late locking 0
[main] light-locker.c:200 (00:09:02):	 lock on suspend 1
[main] light-locker.c:201 (00:09:02):	 lock on lid 0
[main] light-locker.c:202 (00:09:02):	 idle hint 0
[init_session_id] gs-listener-dbus.c:2193 (00:09:02):	 Got session-id: /org/freedesktop/login1/session/_317
[init_session_id] gs-listener-dbus.c:2198 (00:09:02):	 Got sd-session-id: 17
[init_seat_path] gs-listener-dbus.c:2279 (00:09:02):	 Got seat: /org/freedesktop/DisplayManager/Seat0
[gs_listener_delay_suspend] gs-listener-dbus.c:449 (00:09:02):	 Delay suspend
[gs_listener_x11_acquire] gs-listener-x11.c:172 (00:09:02):	 ScreenSaver Registered
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:09:02):	 obj_path=/org/freedesktop/DBus interface=org.freedesktop.DBus method=NameAcquired destination=:1.1327
[listener_dbus_handle_session_message] gs-listener-dbus.c:1010 (00:09:20):	 Received Lock request
[gs_grab_grab_root] gs-grab-x11.c:647 (00:09:20):	 Grabbing the root window
[gs_grab_get_keyboard] gs-grab-x11.c:153 (00:09:20):	 Grabbing keyboard widget=1D4
[gs_grab_get_mouse] gs-grab-x11.c:213 (00:09:20):	 Grabbing mouse widget=1D4
[gs_manager_create_windows_for_screen] gs-manager.c:548 (00:09:20):	 Creating 2 windows for screen 0
[gs_manager_create_window_for_monitor] gs-manager.c:324 (00:09:20):	 Creating window for monitor 0 [1050,0] (2560x1440)
[gs_manager_create_window_for_monitor] gs-manager.c:324 (00:09:20):	 Creating window for monitor 1 [0,0] (1050x1680)
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[gs_manager_timed_switch] gs-manager.c:445 (00:09:20):	 Start switch to greeter timer
[gs_window_xevent] gs-window-x11.c:381 (00:09:20):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:381 (00:09:20):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:369 (00:09:20):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:369 (00:09:20):	 not raising our windows
[window_map_event_cb] gs-manager.c:233 (00:09:20):	 Handling window map_event event
[gs_listener_resume_suspend] gs-listener-dbus.c:513 (00:09:20):	 Resume suspend: fd=16
[manager_maybe_grab_window] gs-manager.c:204 (00:09:20):	 Moving grab to 0x55a13c8963d0
[xorg_lock_smasher_set_active] gs-grab-x11.c:101 (00:09:20):	 No XFree86-Misc extension present
[gs_grab_move_keyboard] gs-grab-x11.c:450 (00:09:20):	 Moving keyboard grab from 1D4 to 540000C
[gs_grab_move_keyboard] gs-grab-x11.c:457 (00:09:20):	 *** doing X server grab
[gs_grab_release_keyboard] gs-grab-x11.c:279 (00:09:20):	 Ungrabbing keyboard
[gs_grab_get_keyboard] gs-grab-x11.c:153 (00:09:20):	 Grabbing keyboard widget=540000C
[gs_grab_move_keyboard] gs-grab-x11.c:479 (00:09:20):	 *** releasing X server grab
[gs_grab_move_mouse] gs-grab-x11.c:395 (00:09:20):	 Moving pointer grab from 1D4 to 540000C
[gs_grab_move_mouse] gs-grab-x11.c:402 (00:09:20):	 *** doing X server grab
[gs_grab_release_mouse] gs-grab-x11.c:320 (00:09:20):	 Ungrabbing pointer
[gs_grab_get_mouse] gs-grab-x11.c:213 (00:09:20):	 Grabbing mouse widget=540000C
[gs_grab_move_mouse] gs-grab-x11.c:425 (00:09:20):	 *** releasing X server grab
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:09:20):	 obj_path=/org/freedesktop/login1 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[gs_window_xevent] gs-window-x11.c:369 (00:09:20):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:369 (00:09:20):	 not raising our windows
[window_map_event_cb] gs-manager.c:233 (00:09:20):	 Handling window map_event event
[gs_listener_resume_suspend] gs-listener-dbus.c:513 (00:09:20):	 Resume suspend: fd=-1
[gs_window_xevent] gs-window-x11.c:369 (00:09:20):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:369 (00:09:20):	 not raising our windows
[window_map_event_cb] gs-manager.c:233 (00:09:20):	 Handling window map_event event
[gs_listener_resume_suspend] gs-listener-dbus.c:513 (00:09:20):	 Resume suspend: fd=-1
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:197 (00:09:20):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:09:20):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:09:20):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[gs_listener_send_lock_session] gs-listener-dbus.c:180 (00:09:21):	 Send lock session
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:09:21):	 obj_path=(null) interface=(null) method=(null) destination=:1.1327
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:09:21):	 obj_path=/org/freedesktop/login1/session/_317 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:09:22):	 obj_path=/org/freedesktop/login1/user/_974 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:09:22):	 obj_path=/org/freedesktop/login1/seat/seat0 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1374 (00:09:22):	 systemd notified ActiveSession 0
[listener_session_switched_cb] gs-monitor.c:298 (00:09:22):	 Session switched: 0
[gs_manager_stop_switch] gs-manager.c:456 (00:09:22):	 Stop switch to greeter timer
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:09:22):	 obj_path=/org/freedesktop/login1/session/_317 interface=org.freedesktop.login1.Session method=Lock destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1359 (00:09:22):	 systemd requested session lock
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:04):	 obj_path=/org/freedesktop/login1/session/_317 interface=org.freedesktop.login1.Session method=Unlock destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1352 (00:10:04):	 systemd requested session unlock
[gs_listener_delay_suspend] gs-listener-dbus.c:449 (00:10:04):	 Delay suspend
[gs_grab_release] gs-grab-x11.c:510 (00:10:04):	 Releasing all grabs
[gs_grab_release_mouse] gs-grab-x11.c:320 (00:10:04):	 Ungrabbing pointer
[gs_grab_release_keyboard] gs-grab-x11.c:279 (00:10:04):	 Ungrabbing keyboard
[xorg_lock_smasher_set_active] gs-grab-x11.c:101 (00:10:04):	 No XFree86-Misc extension present
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:04):	 obj_path=/org/freedesktop/login1 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:04):	 obj_path=/org/freedesktop/login1/session/c16 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:04):	 obj_path=/org/freedesktop/login1/session/_317 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:04):	 obj_path=/org/freedesktop/login1/user/_974 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[xroot_filter] gs-listener-x11.c:124 (00:10:05):	 ScreenSaver stopped
[listener_dbus_handle_session_message] gs-listener-dbus.c:1010 (00:10:14):	 Received Lock request
[gs_grab_grab_root] gs-grab-x11.c:647 (00:10:14):	 Grabbing the root window
[gs_grab_get_keyboard] gs-grab-x11.c:153 (00:10:14):	 Grabbing keyboard widget=1D4
[gs_grab_get_mouse] gs-grab-x11.c:213 (00:10:14):	 Grabbing mouse widget=1D4
[gs_manager_create_windows_for_screen] gs-manager.c:548 (00:10:14):	 Creating 2 windows for screen 0
[gs_manager_create_window_for_monitor] gs-manager.c:324 (00:10:14):	 Creating window for monitor 0 [1050,0] (2560x1440)
[gs_manager_create_window_for_monitor] gs-manager.c:324 (00:10:14):	 Creating window for monitor 1 [0,0] (1050x1680)
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_xevent] gs-window-x11.c:381 (00:10:14):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:381 (00:10:14):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:381 (00:10:14):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:381 (00:10:14):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:369 (00:10:14):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:369 (00:10:14):	 not raising our windows
[window_map_event_cb] gs-manager.c:233 (00:10:14):	 Handling window map_event event
[gs_listener_resume_suspend] gs-listener-dbus.c:513 (00:10:14):	 Resume suspend: fd=16
[manager_maybe_grab_window] gs-manager.c:204 (00:10:14):	 Moving grab to 0x55a13c8969b0
[xorg_lock_smasher_set_active] gs-grab-x11.c:101 (00:10:14):	 No XFree86-Misc extension present
[gs_grab_move_keyboard] gs-grab-x11.c:450 (00:10:14):	 Moving keyboard grab from 1D4 to 5400025
[gs_grab_move_keyboard] gs-grab-x11.c:457 (00:10:14):	 *** doing X server grab
[gs_grab_release_keyboard] gs-grab-x11.c:279 (00:10:14):	 Ungrabbing keyboard
[gs_grab_get_keyboard] gs-grab-x11.c:153 (00:10:14):	 Grabbing keyboard widget=5400025
[gs_grab_move_keyboard] gs-grab-x11.c:479 (00:10:14):	 *** releasing X server grab
[gs_grab_move_mouse] gs-grab-x11.c:395 (00:10:14):	 Moving pointer grab from 1D4 to 5400025
[gs_grab_move_mouse] gs-grab-x11.c:402 (00:10:14):	 *** doing X server grab
[gs_grab_release_mouse] gs-grab-x11.c:320 (00:10:14):	 Ungrabbing pointer
[gs_grab_get_mouse] gs-grab-x11.c:213 (00:10:14):	 Grabbing mouse widget=5400025
[gs_grab_move_mouse] gs-grab-x11.c:425 (00:10:14):	 *** releasing X server grab
[gs_window_xevent] gs-window-x11.c:369 (00:10:14):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:369 (00:10:14):	 not raising our windows
[window_map_event_cb] gs-manager.c:233 (00:10:14):	 Handling window map_event event
[gs_listener_resume_suspend] gs-listener-dbus.c:513 (00:10:14):	 Resume suspend: fd=-1
[manager_maybe_grab_window] gs-manager.c:204 (00:10:14):	 Moving grab to 0x55a13c8969b0
[xorg_lock_smasher_set_active] gs-grab-x11.c:101 (00:10:14):	 No XFree86-Misc extension present
[gs_grab_move_keyboard] gs-grab-x11.c:443 (00:10:14):	 Window 5400025 is already grabbed, skipping
[gs_grab_move_mouse] gs-grab-x11.c:383 (00:10:14):	 Window 5400025 is already grabbed, skipping
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:14):	 obj_path=/org/freedesktop/login1 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[gs_window_xevent] gs-window-x11.c:369 (00:10:14):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:369 (00:10:14):	 not raising our windows
[window_map_event_cb] gs-manager.c:233 (00:10:14):	 Handling window map_event event
[gs_listener_resume_suspend] gs-listener-dbus.c:513 (00:10:14):	 Resume suspend: fd=-1
[gs_window_xevent] gs-window-x11.c:369 (00:10:14):	 not raising our windows
[gs_window_xevent] gs-window-x11.c:369 (00:10:14):	 not raising our windows
[window_map_event_cb] gs-manager.c:233 (00:10:14):	 Handling window map_event event
[gs_listener_resume_suspend] gs-listener-dbus.c:513 (00:10:14):	 Resume suspend: fd=-1
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 0: x=1050 y=0 w=2560 h=1440
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 0: x=1050 y=0 w=2560 h=1440
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:197 (00:10:14):	 got geometry for monitor 1: x=0 y=0 w=1050 h=1680
[update_geometry] gs-window-x11.c:210 (00:10:14):	 using geometry for monitor 1: x=0 y=0 w=1050 h=1680
[gs_window_move_resize_window] gs-window-x11.c:243 (00:10:14):	 Move and/or resize window on monitor 1: x=0 y=0 w=1050 h=1680
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:24):	 obj_path=/org/freedesktop/login1/session/_317 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:26):	 obj_path=/org/freedesktop/login1/session/_317 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[xroot_filter] gs-listener-x11.c:124 (00:10:26):	 ScreenSaver stopped
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:31):	 obj_path=/org/freedesktop/login1/session/_317 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
[listener_dbus_handle_system_message] gs-listener-dbus.c:1343 (00:10:31):	 obj_path=/org/freedesktop/login1/session/_35 interface=org.freedesktop.DBus.Properties method=PropertiesChanged destination=(null)
zsh: terminated  light-locker --debug
```

**Comment by worldofpeace** [comment]
@chrishoage If you look in the issue tracker there's already many duplicates of this exact issue.

**Comment by chrishoage** [comment]
It wasn't clear to me that this was the exact same issue, considering it works for me on the first lock

from what I was reading in the other issues it didn't work for them ever

**Comment by Elliot2560** [comment]
The `dm-tool switch-to-greeter` workaround from #138 worked for me, with the unfortunate side effect that whichever window was active when the screen was locked loses keyboard until I switch to a different window and back.

**Comment by chrishoage** [comment]
I had time to downgrade `systemd` and `systemd-sysvcompat` and the issue is gone. 

There seems to be a compatibility issue with systemd version 243.0-1 and light-locker

**Comment by Ham5andw1ch** [comment]
I too am having this issue. Using downgraded systemd for the time being until we figure out why this is.

**Comment by emanuelserpa** [comment]
I'm having this same issue on Arch Linux

**Comment by YanDoroshenko** [comment]
Seems like I'm having the same issue.

**Comment by irminsul** [comment]
Same here.

**Comment by marcusg** [comment]
same on arch....

**Comment by worldofpeace** [comment]
I you don't have any extra context to add to the issue, an equivalent could be using the thumbs up feature.

**Comment by worldofpeace** [comment]
An easy way to reproduce this would be
```
light-locker-command -l # run this twice
```

Always produces this issue with systemd 243.

It's likely to be a bug in logind https://github.com/systemd/systemd/commits/v243/src/login,
probably could bisect for the issue. I do know in GNOME they had issues with logind in 234 that required an upstream fix https://github.com/systemd/systemd/commit/471cffcfb0e005b7c4044b3b52cc4f25d217efac#diff-0eb89e0717dc797d17bc97f3e4794a6a.

**Comment by worldofpeace** [comment]
Didn't have the convenience to bisect, but reverting https://github.com/systemd/systemd/commit/3b92c086a8d5338e2164ffa0ae48b3d03d10cfb5 got rid of the issue. I'm pretty sure it's what causes the issue here as well.

**Comment by petrmanek** [comment]
After a recent update, I'm also getting this bug. Running light-locker 1.8.0-2 with systemd 243.51-1 on arch (5.3.1.arch1-1). My current workaround is not to run light-locker at all.

**Comment by YanDoroshenko** [comment]
@petrmanek Awesome workaround man. I know a couple more of the same kind - to not run GNU/Linux, to not use a computer and to not exist :D

**Comment by petrmanek** [comment]
> @petrmanek Awesome workaround man. I know a couple more of the same kind - to not run GNU/Linux, to not use a computer and to not exist :D

This almost sounds like a enterprise strategy of sorts. :D :D

But seriously, I don't really like selective package downgrading. So this, albeit a crude solution, makes sense to me as a short-term fix. If this becomes a long-term issue though, my next "workaround" for this will be to find a nice Wayland compositor and cease using LightDM completely.

**Comment by jee-r** [comment]
> I had time to downgrade `systemd` and `systemd-sysvcompat` and the issue is gone.
> 
> There seems to be a compatibility issue with systemd version 243.0-1 and light-locker

I have the same issue on my laptop under Archlinux and  i confirm after downgrading `systemd` and `systemd-sysvcompat` from `243.51-1` to `242.84-2` the issue is gone.

**Comment by pbbob** [comment]
Since this seems to be an issue with systemd, and not with light-locker, has anyone reported this to the systemd folks?

**Comment by chrishoage** [comment]
If you're confident this is an actual issue with systemd and not an issue with how light-locker uses/interfaces with systemd then by all means go file an issue with systemd 

Searching their github issue tracker this issue does not have an active GHI open. 

The reason I started with a GHI in the light-locker repo is because systemd otherwise works fine, and I definitely don't understand what the issue is enough to understand it's an upstream issue.

**Comment by worldofpeace** [comment]
I've declined to open an issue with upstream systemd for the same reasons. Mostly because this looks like a slightly unmaintained project, so it felt inappropriate to make such a report.

However, bringing the issue to them could perhaps help us figure out what code changes need to be make in light-locker, if any, so we can proceed with fixing this.

**Comment by worldofpeace** [comment]
https://github.com/systemd/systemd/issues/13769 is tracking this.

**Comment by newrokor** [comment]
As a work around for those using Arch/XFCE (and possibly others): Replacing light-locker by xfce4-screensaver (incl. locker) works perfect for me using an up-to-date systemd. Maybe due to no switch of the VT?

**Comment by worldofpeace** [comment]
Though note @newrokor that may not work for everyone as their screenlocker has a different dbus api.

**Comment by worldofpeace** [comment]
We got this fixed in nixos https://github.com/NixOS/nixpkgs/issues/69399.
Your distro should include commits from https://github.com/systemd/systemd/pull/13811 into systemd.

**Comment by petrmanek** [comment]
I haven't seen the bug for a while now. Currently running systemd v243.78. Don't know if the commits referenced by @worldofpeace already managed to make it into the mainline since v243.51.

**Comment by rudism** [comment]
This problem still occurs for me on systemd 243.78-2 in arch linux.

**Comment by jee-r** [comment]
> This problem still occurs for me on systemd 243.78-2 in arch linux.

~~For me it's ok i no longer have this issue on my Arch~~
~~`systemd` and `systemd-sysvcompat` in version `243.78-2`~~

~~Did you try to reboot ?~~

edit:

Oh no the #$?%ing lock screen is back :cry:

**Comment by luben93** [comment]
This problem still occurs for me too using systemd 243.78-2 in arch linux.

**Comment by renataogarcia** [comment]
For Arch users - `243.78-2` does not include the fix yet. See [bug report](https://bugs.archlinux.org/task/64439) for more info.

**Comment by Elliot2560** [comment]
The issue is fixed for Arch now in systemd 273.162-2, which includes [this commit](https://github.com/systemd/systemd-stable/commit/6e3cfe2e58f5dd93f66c05c74cda2879ea8a6888) that fixes [this upstream issue](https://github.com/systemd/systemd/issues/13769). I upgraded systemd and tried using light-locker on my two computers and they seem to work fine.

**Comment by petrmanek** [comment]
@Elliot2560  Just upgraded to 243.162-2. Will update the thread if I see anything suspicious.

**Comment by atmarx** [comment]
Worked for me -- thank you!  It's gonna take a few days to break the habit of immediately switching to tty6 and unlocking the session from there :)

**Comment by miniskipper** [comment]
Finally :)

**Comment by tuxayo** [comment]
I wonder how many users switched to another screen locker due to this :(

(which still happening on other distros not having yet the update)

**Comment by tintou** [comment]
So if it is fixed upstream, let's close this issue

**Comment by rafamanzo** [comment]
After upgrading systemd on Arch to 247, the same problem is back in two machines I have with the same setup.

Is anyone else experiencing this again after upgrading?

If so, should we open a new issue to track it or reopen this one?

**Comment by rudism** [comment]
I'm having this (or at least a very similar) issue again after upgrading as well.

**Comment by chrishoage** [comment]
Assuming you're on Arch: The package maintainer cherry picked this commit https://github.com/the-cavalry/light-locker/pull/153 to the build in the arch package repo https://github.com/archlinux/svntogit-community/commit/f2f327b937416353702e29851201d0c24e91ed77

There is this bug tracking it in the Arch bug tracker https://bugs.archlinux.org/task/68905

Downgrading from 1.9.0-3 to 1.9.0-2 works around the issue for now
